R/dynamic_referencing.R

```diff
@@ -0,0 +1,202 @@
+#' Manage dynamic refrerencing
+#' 
+#' The job of the dynamic referencer is to aggregate the references that exist from
+#' specifications, test cases, test code and in the validation report and
+#' update them accordingly. This helps with the painful cases where a new reference
+#' may be added between existing numbering and all subsequent references need to be 
+#' updated
+#' 
+#' The user should not need to use this object directly, it will be called upon 
+#' on rendering of the contents in the validation report.
+#' 
+#' By default, the expected indicator for a reference is "@@", though this can be
+#' changed. A reference starts with the indicator (@@) and can then be any 
+#' contiguous (no white spaces or special characters) alphanumeric sequence, or include 
+#' an underscore or dash. ie. @@THISIS_A-reference1234 is valid for the entire string, 
+#' but @@THISIS_A-reference.12345 is a dynamic reference up to the ".".
+#' 
+#' The method 'scrape_references' takes in a vector of strings, and an indicator whether
+#' the input file text is a specification ('spec'), test case ('test_case') or 
+#' test code ('test_code'). This allows numbering to increase independently for each.
+#' However, every reference must be unique. ie. @@reference whether it shows up in a 
+#' test case or spec will be the same reference.
+#' 
+#' The method 'reference_insertion' takes a vector of strings and replaces references
+#' with their numeric values.
+#' 
+#' 
+#' @keywords internal
+#' @export vt_dynamic_referencer
```

Comment:
Is this function going to be used by a user? Or is it being exported for documentation purposes?

Reply:
so its a situation kind of like the reporter in testthat. Not really intended for most users to be interacting with it, but it needs to be exported